app/models/user.rb

```diff
@@ -344,14 +344,16 @@ def destroy
     super
   end
 
-  %w[military_information payment].each do |emis_method|
-    define_method(emis_method) do
-      emis_model = instance_variable_get(:"@#{emis_method}")
-      return emis_model if emis_model.present?
-
-      emis_model = "EMISRedis::#{emis_method.camelize}".constantize.for_user(self)
-      instance_variable_set(:"@#{emis_method}", emis_model)
-      emis_model
+  def military_information
+    if Flipper.enabled?(:military_information_vaprofile)
+      if @military_information.present?
```

Comment:
I would refactor this to be:

`@military_information ||= FormProfile.new(form_id: nil, user: self).initialize_military_information`

Reply:
Nice. Good call.